core/string/ustring.cpp

```diff
@@ -3176,6 +3176,71 @@ String String::erase(int p_pos, int p_chars) const {
 	return left(p_pos) + substr(p_pos + p_chars);
 }
 
+String String::remove_char(char32_t p_char) const {
```

Comment:
The code between the `char_32_t` and `Vector<char_32_t>` is way too similar. Would it be possible to combine the two somehow to share the same code?

Reply:
Can see, but I don't think it's worth the complexity as it'll not be equivalent, there's a limit to how useful it is to merge code when it's not equivalent, especially as only about half of the lines are actually the same in the second block